### PR TITLE
[#769] updated regex to properly handle <> brackets

### LINKED ISSUE #769: URL for Teams are screwed up
Apprently it works on OpenPaaS and not on Twake calendar-ng

[040000008200E00074C5B7101A82E008000000002EE74DED1F7FDC01000000000000000010000000A83CFA25BB8A754B824BE1C645EEDB80_2026-04-01T08_30_00Z.ics.txt](https://github.com/user-attachments/files/26626680/040000008200E00074C5B7101A82E008000000002EE74DED1F7FDC01000000000000000010000000A83CFA25BB8A754B824BE1C645EEDB80_2026-04-01T08_30_00Z.ics.txt)

Investigate!

### PR DESCRIPTION
**Pull request by Eriikah**
resolves #769 

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for URL detection, validating https/www normalization, trailing-punctuation trimming, parentheses/angle-bracket edge cases, preservation of surrounding text, and correct link rendering.

* **Refactor**
  * Moved URL detection into a shared utility and updated rendering to produce clickable links that open in a new tab safely.

### CODE REVIEW

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

**Review by lethemanh — APPROVED** [review]

**Review by chibenwa — CHANGES_REQUESTED** [review]
No tests ?

**Comment by chibenwa** [comment]
(but good diagnostic ;-) )

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

File: src/components/Event/utils/detectUrls.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    if (lastIndex < offset) {
+      parts.push(
+        <React.Fragment key={lastIndex}>
+          {text.slice(lastIndex, offset)}
+        </React.Fragment>
+      )
+    }
+
+    // Normalize href
+    const href = url.startsWith('http') ? url : `https://${url}`
```
Comment (lethemanh) [inline src/components/Event/utils/detectUrls.tsx]: if the url starts with `https` the condition `url.startsWith('http')` is correct as well, it should:

```suggestion
    const href = url.startsWith('http://') ? url : `https://${url}`
```

File: src/components/Event/utils/detectUrls.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  const parts = []
+  let lastIndex = 0
+
+  text.replace(urlRegex, (match, _, offset: number) => {
+    // Strip trailing punctuation, but preserve balanced parentheses (e.g. Wikipedia)
+    let url = match
+
+    const countChar = (value: string, char: '(' | ')'): number =>
+      (value.match(new RegExp(`\\${char}`, 'g')) || []).length
```
Comment (lethemanh) [inline src/components/Event/utils/detectUrls.tsx]: extract this function

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

File: src/components/Event/utils/detectUrls.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    let url = match
+
+    while (/[.,;:!?)]$/.test(url)) {
+      if (url.endsWith(')')) {
+        const openCount = countChar(url, '(')
+        const closeCount = countChar(url, ')')
+        if (closeCount <= openCount) break
+      }
+      url = url.slice(0, -1)
+    }
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply src/components/Event/utils/detectUrls.tsx:22]: Seems relevant and shall likely be tests, no?

File: src/components/Event/utils/detectUrls.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          {text.slice(lastIndex, offset)}
+        </React.Fragment>
+      )
+    }
+
+    // Normalize href
+    const href =
+      url.startsWith('http://') || url.startsWith('https://')
+        ? url
+        : `https://${url}`
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply src/components/Event/utils/detectUrls.tsx]: Idem

**Comment by Arsnael** [comment]
Docker image published for this PR: linagora/twake-calendar-web:pr-792

**Review by chibenwa — APPROVED** [review]
